ert/data/record/_transformation.py

```diff
@@ -227,3 +229,74 @@ async def _save_record_to_file(record: Record, location: Path, mime: str) -> Non
     else:
         async with aiofiles.open(str(location), mode="wb") as fb:
             await fb.write(record.data)  # type: ignore
+
+
+class RecordTreeTransformation(RecordTransformation):
+    """
+    Write all leaf records of a RecordTree to individual files.
+    """
+
+    def __init__(self, sub_path: Optional[str] = None):
+        if sub_path is not None:
+            raise NotImplementedError
+
+    async def transform_input(
+        self,
+        record: Record,
+        mime: str,
+        runpath: Path,
+        location: Path,
+    ) -> None:
+        if isinstance(record, NumericalRecordTree):
+            for key, leaf_record in record._flat_record_dict.items():
+                location_key = f"{key}-{location}"
+                await get_serializer(mime).encode_to_path(
+                    leaf_record.data, path=runpath / location_key
+                )
+        else:
+            raise TypeError("Only NumericalRecordTrees can be transformed.")
+
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        raise NotImplementedError
+
+
+class EclSumTransformation(RecordTreeTransformation):
+    """A transformation that will transform binary output from Eclipse into a
+    NumericalRecordTree.
+
+    Input, as a NumericalRecordTree, can be transformed into multiple files
+    of a serializable format."""
+
+    def __init__(self, smry_keys: Optional[List[str]]):
```

Comment:
Why are the keys optional?

Reply:
I think we want to implement that `None` will result in all smry keys being imported. Changing the `ValueError` to a `NotImplementedError` when `None` is supplied.

Reply:
No, that is probably not a good idea, as `smry_keys!=None` in `_builder.py` means that we should employ the `EclSumTransformation`. Removing `Optional`.

Reply:
Yeah, at this abstraction level, we should be explicit. All should have some other representation than None.